backend/pkg/web/handler/resource_related.go

```diff
@@ -93,3 +95,104 @@ func CreateRelatedResources(c *gin.Context) {
 	c.JSON(http.StatusOK, gin.H{"success": true, "data": summary, "source": fastenSourceCredential})
 
 }
+
+func EncounterUnlinkResource(c *gin.Context) {
```

Comment:
Does this function still work if `Encounter` -> `X` -> `Y`

A comment describing the usecase for this function would be helpful.

Reply:
Also, this function seems to be pretty complex, not sure all this logic should be in the hander --> might be better to put it in the database access layer, with robust tests.

Reply:
Fantastic, thanks for confirming!